### PR TITLE
fix(oomstore): stream push processor not exit graceful

### LINKED ISSUE #952: bug:  offline has no stream feature.
After pushing the stream feature, the offline store has no stream feature.

Reproduction steps：

1. apply this meta
```yaml
items:
  - kind: Entity
    name: device
    description: device
    groups:
      - name: phone
        category: batch
        description: phone
        features:
          - name: price
            value-type: int64
            description: price
          - name: model
            value-type: string
            description: model
  - kind: Entity
    name: user
    description: user
    groups:
      - name: student
        category: batch
        description: student
        features:
          - name: age
            value-type: int64
            description: age
      - name: user-click
        category: stream
        description: user click post feature
        features:
          - name: last_5_click_posts
            value-type: string
            description: user last 5 click posts
          - name: number_of_user_started_posts
            value-type: int64
            description: number of posts that users stared today

```

2. run push:
```
$ oomcli push --entity-key 1 --group user-click --features last_5_click_posts=1,2,3,4,5 --features number_of_user_started_posts=10
```

3. check db if has offline stream table

```shell
(base) ➜  oomcli git:(main) ✗ pgcli -h localhost -p 25432 -u postgres
Password for postgres:
Server: PostgreSQL 14.0
Version: 3.2.0
Home: http://pgcli.com
postgres@localhost:postgres> \c oomstore ;
You are now connected to database "oomstore" as user "postgres"
Time: 0.028s
postgres@localhost:oomstore> \dt ;
+----------+------------------------+--------+----------+
| Schema   | Name                   | Type   | Owner    |
|----------+------------------------+--------+----------|
| public   | entity                 | table  | postgres |
| public   | feature                | table  | postgres |
| public   | feature_group          | table  | postgres |
| public   | feature_group_revision | table  | postgres |
| public   | online_stream_3        | table  | postgres |
+----------+------------------------+--------+----------+
SELECT 6
Time: 0.020s
postgres@localhost:oomstore>

```

### PR DESCRIPTION
**Pull request by lianxmfor**
close #952

### CODE REVIEW

File: pkg/oomstore/cache.go [hunk 1]
```diff
@@ -28,9 +28,10 @@ type GroupBuffer struct {
 }
 
 type StreamPushProcessor struct {
-	capacity  int
-	minPeriod time.Duration
-	quit      chan bool
+	capacity          int
+	minPeriod         time.Duration
+	quit              chan bool
+	waitProcessorQuit chan struct{}
```
Comment (wfxr) [inline pkg/oomstore/cache.go:34]: Are both `quit` and `waitProcessorQuit` required ?

Comment (wfxr) [reply pkg/oomstore/cache.go:34]: Or `notifyQuit` and `waitQuit`? BTW can we use the same type (choose one of `bool` or `struct{}`) ?

Comment (lianxmfor) [reply pkg/oomstore/cache.go:34]: sound good! I will use `struct{}` since it doesn't need memory.

**Review by wfxr — APPROVED** [review]
LGTM!

**Pull request merged**